cpp/include/Ice/ServerAuthenticationOptions.h

```diff
@@ -0,0 +1,415 @@
+         *
+         * Example of setting serverCertificateSelectionCallback:
+         * ```cpp
+         * PCCERT_CONTEXT _serverCertificate = ...; // Load the server certificate using WinCrypt API
+         *
+         * communicator->createObjectAdapterWithEndpoints(
+         *   "Hello",
+         *   "ssl -h 127.0.0.1 -p 10000",
+         *   ServerAuthenticationOptions {
+         *     .serverCertificateSelectionCallback = [this](const std::string&) {
+         *        // Increment the certificate context reference count to ensure it remains
+         *        // valid for the duration of the connection. The SSL transport will release
+         *        // it after closing the connection.
+         *        CertDuplicateCertificateContext(_serverCertificate);
+         *        return _serverCertificate;
+         *      }
+         * };
+         * ...
+         * CertFreeCertificateContext(_serverCertificate); // Release the certificate when no longer needed
+         * ```
+         *
+         * See Detailed Wincrypt documentation for [CERT_CONTEXT](
+         * https://learn.microsoft.com/en-us/windows/win32/api/wincrypt/ns-wincrypt-cert_context) and
+         * [CertCreateCertificateContext](https://learn.microsoft.com/en-us/windows/win32/api/wincrypt/nf-wincrypt-certcreatecertificatecontext)
+         */
+        std::function<PCCERT_CONTEXT(const std::string& host)> serverCertificateSelectionCallback;
+
+        /**
+         * The trusted root certificates used for validating the client's certificate chain. If this field is set, the
+         * server's certificate chain is validated against these certificates; otherwise, the system's default root
+         * certificates are used.
+         *
+         * @remarks The trusted root certificates are only used by the default validation callback, they are ignored by
+         * custom validation callbacks set with clientCertificateValidationCallback.
+         *
+         * The application must ensure that the certificate store remains valid during the setup of the ObjectAdapter.
+         * It is also the application's responsibility to release the certificate store object after the ObjectAdapter
+         * has been created to prevent memory leaks.
+         *
+         * Example of setting trustedRootCertificates:
+         * ```cpp
+         * HCERTSTORE rootCerts = ...; // Populate with X.509 certificates
+         *
+         * communicator->createObjectAdapterWithEndpoints(
+         *   "Hello",
+         *   "ssl -h 127.0.0.1 -p 10000",
+         *   ServerAuthenticationOptions { .trustedRootCertificates = rootCerts; }
+         * });
+         *
+         * auto communicator = Ice::initialize(initData);
+         * CertCloseStore(rootCerts); // It is safe to close the rootCerts store now.
+         * ```
+         */
+        HCERTSTORE trustedRootCertificates;
+
+        /**
+         * Whether or not client-side authentication is required.
```

Comment:
Whether or a not the client must provide a certificate.

I would also put `= false` or `= true` to make the default obvious.